wgengine/router/router_linux.go

```diff
@@ -987,25 +1022,43 @@ func cidrDiff(kind string, old map[netaddr.IPPrefix]bool, new []netaddr.IPPrefix
 		ret[cidr] = true
 	}
 
+	var delFail []error
 	for cidr := range old {
 		if newMap[cidr] {
 			continue
 		}
 		if err := del(cidr); err != nil {
 			logf("%s del failed: %v", kind, err)
-			return ret, err
+			delFail = append(delFail, err)
+		} else {
+			delete(ret, cidr)
 		}
-		delete(ret, cidr)
 	}
+	if len(delFail) == 1 {
+		return ret, delFail[0]
+	}
+	if len(delFail) > 0 {
+		return ret, fmt.Errorf("%d delete %s failures; first was: %w", len(delFail), kind, delFail[0])
+	}
+
+	var addFail []error
```

Comment:
If we're going down the path of trying to soldier on despite errors, should we also try to add stuff even if deletes failed?

I'm not a fan of either because it leads to states that are going to turn into nasty bug reports, but if we're going to do it, let's go all-in?

Reply:
I couldn't reason easily about the safety of that, so I'd prefer not. It seems like some adds may be dependents on deletes finishing.

But with making delete idempotent now, it seems we'll basically never get stuck on delete now. If we do, it's worth stopping.